### PR TITLE
[MNT] Removes coverage upload steps from CI

### LINKED ISSUE #6858: [MNT] removal of coverage upload steps from CI
https://github.com/sktime/sktime/pull/6363 removed coverage reporting from CI that runs on al pull requests. But the steps to upload coverage report is not removed, and hence all PR now report a high number of warnings.

### PR DESCRIPTION
**Pull request by Prtm2110**
#### Reference Issues/PRs
Fixes #6858 

#### Did you add any tests for the change?
No

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Comment by fkiraly** [comment]
FYI @yarnabrina

**Review by yarnabrina — CHANGES_REQUESTED** [review]
Hi @Prtm2110, thank you for your contribution.

This is a good start, but it is incomplete. We have multiple GHA workflows, and you have removed it from just one. I think we should do for all of them.

@fkiraly if we remove coverage report altogether to improve CI performance, is that good/okay? Should we consider some alternative way to get these information? Should we also modify README to avoid showing 0% or similar?

**Comment by Prtm2110** [comment]
> We have multiple GHA workflows, and you have removed it from just one. I think we should do for all of them.

Okay, I have made the necessary changes.

**Comment by fkiraly** [comment]
> @fkiraly if we remove coverage report altogether to improve CI performance, is that good/okay? Should we consider some alternative way to get these information? Should we also modify README to avoid showing 0% or similar?

From `test-all` runs maybe?

**Comment by fkiraly** [comment]
I think badges can be made to default to a tag, and the last codecov report of that tag, but I don't have the details. Maybe one of the chatbots can help?

**Comment by Prtm2110** [comment]
> I think badges can be made to default to a tag, and the last codecov report of that tag, but I don't have the details. Maybe one of the chatbots can help?

Shall I proceed?

**Comment by fkiraly** [comment]
That would be great!

**Comment by Prtm2110** [comment]
Sorry for the inactivity. I have made the necessary changes. Please let me know if anything else is required.

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ sktime is a library for time series analysis in Python. It provides a unified in
 | **Code** |  [![!pypi](https://img.shields.io/pypi/v/sktime?color=orange)](https://pypi.org/project/sktime/) [![!conda](https://img.shields.io/conda/vn/conda-forge/sktime)](https://anaconda.org/conda-forge/sktime) [![!python-versions](https://img.shields.io/pypi/pyversions/sktime)](https://www.python.org/) [![!black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)  |
 | **Downloads** | ![PyPI - Downloads](https://img.shields.io/pypi/dw/sktime) ![PyPI - Downloads](https://img.shields.io/pypi/dm/sktime) [![Downloads](https://static.pepy.tech/personalized-badge/sktime?period=total&units=international_system&left_color=grey&right_color=blue&left_text=cumulative%20(pypi))](https://pepy.tech/project/sktime) |
 | **Citation** | [![!zenodo](https://zenodo.org/badge/DOI/10.5281/zenodo.3749000.svg)](https://doi.org/10.5281/zenodo.3749000) |
+| **Coverage** | [![Coverage Status](https://img.shields.io/codecov/c/github/sktime/sktime/main?logo=codecov&tag=latest)](https://codecov.io/gh/sktime/sktime/branch/main) |
```
Comment (fkiraly) [inline README.md:22]: Nice! Could we move this to the "code" row and replace the "black" badge?

Comment (Prtm2110) [reply README.md]: Sure.

**Comment by fkiraly** [comment]
@yarnabrina, I have re-requested a review

File: README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ sktime is a library for time series analysis in Python. It provides a unified in
...
-| **Code** |  [![!pypi](https://img.shields.io/pypi/v/sktime?color=orange)](https://pypi.org/project/sktime/) [![!conda](https://img.shields.io/conda/vn/conda-forge/sktime)](https://anaconda.org/conda-forge/sktime) [![!python-versions](https://img.shields.io/pypi/pyversions/sktime)](https://www.python.org/) [![!black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)  |
+| **Code** |  [![!pypi](https://img.shields.io/pypi/v/sktime?color=orange)](https://pypi.org/project/sktime/) [![!conda](https://img.shields.io/conda/vn/conda-forge/sktime)](https://anaconda.org/conda-forge/sktime) [![!python-versions](https://img.shields.io/pypi/pyversions/sktime)](https://www.python.org/) [![Coverage Status](https://img.shields.io/codecov/c/github/sktime/sktime/main?logo=codecov&tag=latest)](https://codecov.io/gh/sktime/sktime/branch/main)  |
```
Comment (yarnabrina) [inline README.md:19]: I'm not really a fan of this badge. It's reporting 81% from the last time coverage was uploaded, which as of now is 4 months old. This is not representative of current test coverage, and potentially misleading.

@fkiraly if we are not uploading coverage to make CI faster, do we need to report coverage, or can we skip?

**Review by yarnabrina — CHANGES_REQUESTED** [review]
Blocking for now to discuss outdated badge status.

@Prtm2110 can you please update your branch with the main branch also? I doubt CI files will be affected much, but still it's good to be updated.

**Comment by Prtm2110** [comment]
The update `[AUTOMATED] update CONTRIBUTORS.md (https://github.com/sktime/sktime/pull/7012/commits/f02a99d562b81b32ee8d1a7d152a0455ae569892)` is intended, right? Or did I do something wrong?

File: README.md (hunk 2 above)
Comment (fkiraly) [reply README.md:19]: hm, I thought thet test-all runs would upload coverage?

If this is not updated, then let's remove the badge.

**Comment by fkiraly** [comment]
> The update `[AUTOMATED] update CONTRIBUTORS.md (https://github.com/sktime/sktime/pull/7012/commits/f02a99d562b81b32ee8d1a7d152a0455ae569892)` is intended, right? Or did I do something wrong?

Yes, this automatically updates based on changes to the `.allcontributor-src` file.

**Comment by Prtm2110** [comment]
Are there any additional modifications to be implemented?

**Review by yarnabrina — DISMISSED** [review]

**Comment by Prtm2110** [comment]
Apologies for the inactivity. I will address the failing tests as soon as possible and commit the changes.

**Pull request merged**